eliHenderson/assignments/fastSlow/lc141.js

```diff
@@ -0,0 +1,69 @@
+// Problem Statement #
+
+// Given the head of a Singly LinkedList, write a function to determine if the LinkedList has a cycle in it or not.
+/**
+ * process: two pointer
+ * one faster pointer and one slow pointer
+ * 1 => 2=> 3=> 4=>5=>6=>3
+ *
+ * set both pointer as the head node
+ * increment them from the head
+ *
+ * while loop
+ *  keep incrementing until either
+ *  one node reaches null or
+ *  the two nodes meet
+ *
+ * time complexity: O(N)
+ * space complexity: O(1)
+ *
+ */
+
+class Node {
+  constructor(value, next = null) {
+    this.value = value;
+    this.next = next;
+  }
+}
+
+const has_cycle = function (head) {
+  let slow = head,
+    fast = head;
+
+  while (fast !== null && fast.next !== null) {
+    fast = fast.next.next; // 2x faster than slow
+    slow = slow.next;
+
+    if (fast === slow) return true; // if the nodes meet, return true
+  }
+
+  return false;
+};
+
+const has_cycle = function (head) {
+  let slow = head,
+    fast = head;
+
+  while (fast !== null && fast.next !== null) {
+    fast = fast.next.next;
+    slow = slow.next;
+
+    if (fast === slow) return true;
+  }
+
+  return false;
+};
```

Comment:
If it's nice, write it twice! Great job demonstrating that and building your muscle memory :100: